### PR TITLE
use QCoreApplication for Oven when headless version is used

### PR DESCRIPTION
**Pull request by birarda**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7336/app_12447_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7337/app_12447_PR.apk

File: tools/oven/src/Oven.cpp [hunk 1]
```diff
@@ -47,41 +29,27 @@ Oven::Oven(int argc, char* argv[]) :
...
 
 Oven::~Oven() {
-    // cleanup the worker threads
-    for (auto i = 0; i < _workerThreads.size(); ++i) {
-        _workerThreads[i]->quit();
-        _workerThreads[i]->wait();
+    // quit and wait on the worker threads
+    for (auto& thread : _workerThreads) {
+        thread->quit();
+        thread->wait();
```
Comment (Atlante45) [inline tools/oven/src/Oven.cpp]: Would probably be best to quit the all first and then wait on them.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7338/app_12447_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7338/HighFidelity-Beta-PR12447-7ee6eead52ec366fdeb869825d9b9ddca2276e44.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7344/app_12447_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7344/HighFidelity-Beta-PR12447-2e466dc76cc852a2f21d108f36949cda0a2c35a8.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7344/HighFidelity-Beta-PR12447-2e466dc76cc852a2f21d108f36949cda0a2c35a8.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7362/app_12447_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7363/app_12447_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7363/HighFidelity-Beta-PR12447-79f5d0cee21161054b12d220b28d1d4bac546887.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7363/HighFidelity-Beta-PR12447-79f5d0cee21161054b12d220b28d1d4bac546887.exe

**Comment by birarda** [comment]
#### Test Plan

Run the following tests:

- https://highfidelity.testrail.net/index.php?/cases/view/998
- https://highfidelity.testrail.net/index.php?/cases/view/3481
- https://highfidelity.testrail.net/index.php?/cases/view/3480

**Comment by birarda** [comment]
I moved this to RC65 since it causing more than just the dock bounce annoyance on OS X. Also breaks baking on DigitalOcean domains since the oven is attempting to load libraries needed by QtGui that aren't present.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7769/app_12447_PR.apk

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7783/app_12447_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7783/HighFidelity-Beta-PR12447-79f5d0cee21161054b12d220b28d1d4bac546887.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7783/HighFidelity-Beta-PR12447-79f5d0cee21161054b12d220b28d1d4bac546887.dmg

**Comment by qlimnavarro** [comment]
Currently Testing.

**Comment by qlimnavarro** [comment]
Pass - @conklin94122

**Pull request merged**